tutorials/Oracles/Oracles.ipynb

```diff
@@ -0,0 +1,1015 @@
+    "$$= \\frac{1}{\\sqrt{2}} \\big(U_{7,phase} |110\\rangle + U_{7,phase} |111\\rangle\\big) = \\frac{1}{\\sqrt{2}} \\big(|110\\rangle - |111\\rangle\\big) := |\\gamma\\rangle$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now the question is if our input state was modified during this operation; lets simplify $|\\gamma\\rangle$:\n",
+    "$$|\\gamma\\rangle = \\frac{1}{\\sqrt{2}} \\big(|110\\rangle - |111\\rangle\\big) = |11\\rangle \\otimes \\frac{1}{\\sqrt{2}} \\big(|0\\rangle - |1\\rangle\\big) = $$\n",
+    "$$= |11\\rangle \\otimes |-\\rangle = |11-\\rangle \\neq |\\beta\\rangle$$\n",
+    "\n",
+    "Here we see that the oracle modifies the input.  This is conditioned on the input state being a *superposition* of the basis states of the oracle - as a phase oracle will only modify the sign of its basis states.\n",
+    "\n",
+    "> It is also worth noting that while the oracle modified the input when provided a superposition state, it did *not* modify the norm of that state.  As an exercise, you can verify this yourself by taking the norm of $|\\beta\\rangle$ and $|\\gamma\\rangle$, which both will result in a value of $1$.\n",
+    ">\n",
+    ">\n",
+    "> As another exercise, consider how you could distinguish between the input and output state programmatically?  Is there an operation that you could apply to the initial state $|\\beta\\rangle$ and the final state $|\\gamma\\rangle$ to show that the two states are not equivalent?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Marking Oracles\n",
+    "\n",
+    "A marking oracle $U_{mark}$ is an oracle that when provided some state $|x\\rangle$ and some qubit $y$ performs addition modulo 2 between $f(x)$ and $y$.  Hence $U_{mark}$ is an operator that performs the following operation:\n",
+    "\n",
+    "$$U_{mark}|x\\rangle |y\\rangle = U_{mark}\\big(|x\\rangle \\otimes |y\\rangle\\big) = |x\\rangle \\otimes |y \\oplus f(x)\\rangle = |x\\rangle |y \\oplus f(x)\\rangle$$\n",
+    "\n",
+    "#### <span style=\"color:blue\">Demo 2</span>: Marking quantum oracle - alternating bit pattern!\n",
+    "Consider the following marking oracle $U_{alt,mark}$:\n",
+    "* $U_{alt,mark} |x\\rangle|y\\rangle = |x\\rangle|y\\oplus f(x)\\rangle$\n",
+    "* $f(x) = 1$ if $x = |101\\rangle$ or $|x\\rangle = |010\\rangle$\n",
+    "* $f(x) = 0$ otherwise\n",
+    "* $|x\\rangle$ is a state composted of $3$ qubits\n",
+    "* $|y\\rangle$ is a single qubit"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "open Microsoft.Quantum.Arrays;\n",
+    "open Microsoft.Quantum.Canon;\n",
+    "open Microsoft.Quantum.Intrinsic;\n",
+    "open Microsoft.Quantum.Diagnostics;\n",
+    "\n",
+    "operation Alternating_Marking_Oracle(x: Qubit[], y: Qubit) : Unit\n",
```

Comment:
I would make this demo code shorter as well using ControlledOnBitString([true, false, true], X) and similar for the other pattern